File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/dto/response/SubmissionFullResponseDTO.kt
package io.leonfoliveira.judge.api.dto.response

import io.leonfoliveira.judge.core.domain.entity.Submission
import io.leonfoliveira.judge.core.domain.enumerate.Language
import java.time.LocalDateTime
import java.util.UUID

data class SubmissionFullResponseDTO(
    val id: UUID,
    val member: MemberPublicResponseDTO,
    val problem: ProblemPublicResponseDTO,
    val language: Language,
    val status: Submission.Status,
    val code: AttachmentResponseDTO,
    val createdAt: LocalDateTime,
)

fun Submission.toFullResponseDTO(): SubmissionFullResponseDTO {
    return SubmissionFullResponseDTO(
        id = id,
        member = member.toPublicResponseDTO(),
        problem = problem.toPublicResponseDTO(),
        language = language,
        status = status,
        code = code.toResponseDTO(),
        createdAt = createdAt,
    )
}

File: applications/webapp/src/core/port/driven/repository/dto/response/submission/SubmissionFullResponseDTO.ts
import { Language } from "@/core/domain/enumerate/Language";
import { SubmissionAnswer } from "@/core/domain/enumerate/SubmissionAnswer";
import { SubmissionStatus } from "@/core/domain/enumerate/SubmissionStatus";
import { Attachment } from "@/core/domain/model/Attachment";
import { MemberFullResponseDTO } from "@/core/port/driven/repository/dto/response/member/MemberFullResponseDTO";
import { ProblemFullResponseDTO } from "@/core/port/driven/repository/dto/response/problem/ProblemFullResponseDTO";

export type SubmissionFullResponseDTO = {
  id: string;
  problem: ProblemFullResponseDTO;
  member: MemberFullResponseDTO;
  language: Language;
  status: SubmissionStatus;
  answer: SubmissionAnswer;
  code: Attachment;
  createdAt: string;
};
